### PR TITLE
create new vsc modification

### PR DESCRIPTION
**Pull request by jamal-khey**

### CODE REVIEW

File: src/main/java/org/gridsuite/modification/server/dto/DTOUtils.java [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+
+    public static List<ReactiveCapabilityCurveModificationInfos> convertToReactiveCapabilityCurveModificationInfos(List<ReactiveCapabilityCurveModificationEmbeddable> rCCpoints) {
+        List<ReactiveCapabilityCurveModificationEmbeddable> pointsEmbeddable = !CollectionUtils.isEmpty(rCCpoints) ? rCCpoints : null;
+        return pointsEmbeddable != null ? rCCpoints
+                .stream()
+                .map(value -> new ReactiveCapabilityCurveModificationInfos(value.getQminP(), value.getOldQminP(),
+                        value.getQmaxP(), value.getOldQmaxP(),
+                        value.getP(), value.getOldP()))
+                .toList() : null;
+    }
```
Comment (YenguiSeddik) [inline src/main/java/org/gridsuite/modification/server/dto/DTOUtils.java:23]:
```suggestion
    }public static List<ReactiveCapabilityCurveModificationInfos> convertToReactiveCapabilityCurveModificationInfos(List<ReactiveCapabilityCurveModificationEmbeddable> rCCpoints) {
        return CollectionUtils.isEmpty(rCCpoints) ? List.of() : rCCpoints
                .stream()
                .map(value -> new ReactiveCapabilityCurveModificationInfos(value.getQminP(), value.getOldQminP(),
                        value.getQmaxP(), value.getOldQmaxP(),
                        value.getP(), value.getOldP()))
                .toList();
    }
```

File: src/main/java/org/gridsuite/modification/server/modifications/BatteryModification.java [hunk 2]
```diff
@@ -9,14 +9,7 @@
...
 import com.powsybl.commons.reporter.TypedValue;
-import com.powsybl.iidm.network.Battery;
-import com.powsybl.iidm.network.MinMaxReactiveLimits;
-import com.powsybl.iidm.network.MinMaxReactiveLimitsAdder;
-import com.powsybl.iidm.network.Network;
-import com.powsybl.iidm.network.ReactiveCapabilityCurve;
-import com.powsybl.iidm.network.ReactiveCapabilityCurveAdder;
-import com.powsybl.iidm.network.ReactiveLimits;
-import com.powsybl.iidm.network.ReactiveLimitsKind;
+import com.powsybl.iidm.network.*;
```
Comment (lilian-houdelet) [inline src/main/java/org/gridsuite/modification/server/modifications/BatteryModification.java:12]: maybe you shouldn't use wildcard

File: src/main/java/org/gridsuite/modification/server/modifications/VscModification.java [hunk 3]
```diff
@@ -0,0 +1,302 @@
...
+        String errorMessage = "Converter station '" + converterStationModificationInfos.getEquipmentId() + "' : ";
+        if (vscConverterStation.getReactiveLimits().getKind() == ReactiveLimitsKind.MIN_MAX && (converterStationModificationInfos.getMinimumReactivePower() != null || converterStationModificationInfos.getMaximumReactivePower() != null)) {
+            MinMaxReactiveLimits minMaxReactiveLimits = vscConverterStation.getReactiveLimits(MinMaxReactiveLimits.class);
+            ModificationUtils.getInstance().checkMaxReactivePowerGreaterThanMinReactivePower(minMaxReactiveLimits, converterStationModificationInfos.getMinimumReactivePower(), converterStationModificationInfos.getMaximumReactivePower(), MODIFY_VSC_ERROR, errorMessage);
+        }
+
+        List<ReactiveCapabilityCurveModificationInfos> modificationPoints = converterStationModificationInfos.getReactiveCapabilityCurvePoints();
+        if (modificationPoints != null) {
+            ModificationUtils.getInstance().checkMaxQGreaterThanMinQ(modificationPoints, MODIFY_BATTERY_ERROR, errorMessage);
+        }
```
[the remark this answers is not in the document]
Comment (jamal-khey) [reply src/main/java/org/gridsuite/modification/server/modifications/VscModification.java]: done

File: src/main/java/org/gridsuite/modification/server/modifications/BatteryModification.java (hunk 2 above)
Comment (jamal-khey) [reply src/main/java/org/gridsuite/modification/server/modifications/BatteryModification.java]: done

**Review by lilian-houdelet — APPROVED** [review]

**Review by YenguiSeddik — APPROVED** [review]

**Pull request merged**